### ISSUE TITLE
AdNauseam filter also blocking non-ad content on mac-torrent-download.net

### ISSUE THREAD

**Issue by MrDemocracy** [dhowe/AdNauseam#1230]
### Describe the issue
When the `##DIV[class="ad-box"]`_(line 107)_ filter is used on [mac-torrent-download.net](http://mac-torrent-download.net), it also blocks the real download button.

### One or more specific URLs where the issue occurs
http://mac-torrent-download.net/download.php?aHR0cDovL2NlZXN0eS5jb20vcTlSMTlP

### Screenshot in which the issue can be seen
**With AdNauseam filter list enabled:**
<img width="1280" src="https://user-images.githubusercontent.com/23060864/31316638-775e5cde-ac31-11e7-9e9e-78d863ec2f31.png">

**With AdNauseam filter list disabled:**
<img width="1280" src="https://user-images.githubusercontent.com/23060864/31316637-6b7799ee-ac31-11e7-9fee-666818f6330a.png">

### Steps for anyone to reproduce the issue
Have the AdNauseam filter list enabled.
Try to download something [mac-torrent-download.net](http://mac-torrent-download.net).

### Your settings

- OS/version: macOS 10.13 (17A405)
- Browser/version: Google Chrome 61.0.3163.100 (Official Build) (64-bit)
- AdNauseam version: v3.4.109

**Comment by cqx931** [comment]
Hi @MrDemocracy, this is not an adnauseam specific issue.
According to their website, you should disable all ad-blockers if you want to download anything from the site. The site is intentionally designed in this way, so that you need to see the ad before you download.

http://mac-torrent-download.net/how-to-download-torrents/
![image](https://user-images.githubusercontent.com/3056154/31506058-624f1ffc-af44-11e7-91f2-6f5baf16f7ac.png)

**Comment by dhowe** [comment]
Does uBlock get the same result? If so, then it is a filter list issue, not an AdNauseam issue (as @cqx931 suggests above)

**Comment by cqx931** [comment]
uBlock also hide the download link, but through blocking.
`##DIV[class="ad-box"]`This specific filter user mentioned is from adnauseam filter. But I don't think it much sense to make an exception for this site or trying to separate the ad from the download link.

**Comment by dhowe** [comment]
Agreed

**Comment by MrDemocracy** [comment]
No, don't close the issue. That is fixed by allowing scripts from `www.googletagmanager.com`. 

The only thing still messing this up is the AdNauseam `##DIV[class="ad-box"]`. 
Can't you just change the filter to `zol.com.cn##DIV[class="ad-box"]`?

**Comment by dhowe** [comment]
any objection to the proposed fix, @cqx931 ?

**Comment by MrDemocracy** [comment]
A filter for only blocking the ads and not the download button in the `.ad-box`-container, could be `##.ad-box > .img-center > div:has(script)`

**Comment by cqx931** [comment]
@MrDemocracy  I have just made the changes, please update the adnauseam filter and see whether the issue is fixed for you.
1.`zol.com.cn##DIV[class="ad-box"]` as you proposed
2.`##.ad-box > .img-center > div:has(script)` This rule doesn't work well for me, but I tried with another one: `mac-torrent-download.net##.ad-box > a > img`.

Please also let me know if you see any visible ad that I miss.
Thanks!

**Comment by MrDemocracy** [comment]
`mac-torrent-download.net##.ad-box > a > img` didn't block these ads, so I used `##.ad-box > .img-center > div:has(script)`:
<img width="1280" src="https://user-images.githubusercontent.com/23060864/31513091-67ce8e1a-af8d-11e7-9bd1-c50440fb608b.png">

Here are some filters for other ads that was not blocked:
```
mac-torrent-download.net##.ad-box > .textwidget:has(p > script)
mac-torrent-download.net##.ad-kijisita
mac-torrent-download.net###topnews > .ad-itiran
mac-torrent-download.net##article .round-btn-parent.ad-box
mac-torrent-download.net##.hta3rdpOverlay
||29d65cebb82ef9f.com/*$script,domain=mac-torrent-download.net
||vebadu.com/apu.php$domain=mac-torrent-download.net
||www.fcrgzqkbtgu.co^$domain=mac-torrent-download.net
```

**Comment by friendlyanon** [comment]
Blocking non-ad content here: https://github.com/Ruzzz/uSequential/tree/master/src
with the `##*[id*="ad1"]` rule

**Comment by MrDemocracy** [comment]
@friendlyanon I think you have open a new issue regarding this since it's not related to non-ad content being blocked on **[mac-torrent-download.net](http://mac-torrent-download.net)**.

The `##*[id*="ad1"]` filter is for "pandora", according to the `AdNauseam filters`-list, but I'm not sure if it's meant for [pandora.com](https://www.pandora.com/)(music) or [pandora.net](pandora.net)(rings and jewelry). 
I have browsed both of those sites now, but it doesn't look like  `##*[id*="ad1"]` is being triggered by any elements on any of the pandora's.

**Comment by dhowe** [comment]
@MrDemocracy thanks for the rules above -- do you want to submit a PR [here](https://github.com/dhowe/uAssets) ?

**Comment by dhowe** [comment]
> Blocking non-ad content here: https://github.com/Ruzzz/uSequential/tree/master/src
with the `##*[id*="ad1"]` rule

@friendlyanon that rule was fixed a while back... Can you try updating your adnauseam rules (click the clock and update button next to the adnuseam.txt list) and see if the problem persists?

**Comment by friendlyanon** [comment]
@dhowe forced an update and indeed it works now. I wonder how that occured when I had the auto update checkbox checked.

**Comment by dhowe** [comment]
I believe the period for checks in uBlock is quite long, so it may not have run yet...

**Comment by cqx931** [comment]
Please check https://github.com/dhowe/uAssets/pull/74, this should hide all other ads.

**Comment by dhowe** [comment]
pls re-open if this recurs
